rfc/php_technical_committee.wiki

```diff
@@ -0,0 +1,184 @@
+====== PHP RFC: PHP Technical Committee ======
+  * Version: 0.2
+  * Date: 2023-03-03
+  * Author: Jakub Zelenka, bukka@php.net
+  * Status: Draft
+
+===== Introduction =====
+
+PHP uses its voting system to decide about changes to PHP. This was created mainly for user facing changes. It works
+well for that purpose but it is not that effective for deciding about purely technical changes that impact PHP
+internals and extension API's. It is not clear how it should resolve conflicts on the technical basis between
+contributors that sometimes happen.
+
+
+===== Proposal =====
+
+This RFC proposes an introduction of the PHP Technical Committee (TC) that would decide about all technical aspects of
+php-src. Its primary but not only role is to resolve disagreements between contributors.
+
+==== Definitions ====
+
+The text contains various terms that need further clarification to prevent future wording dispute. Here are their
+definitions:
+
+  * php-src - Source code in https://github.com/php/php-src/.
+  * contributor - Anyone who has already contributed to php-src.
+  * core developer - Anyone with write commit access to php-src.
+  * dev branch - php-src branch consisting of PHP- prefix and major and minor version number (e.g. PHP-8.2).
+  * maintainer - person responsible for extension or SAPI that is listed in the php-src EXTENSION file.
```

Comment:
Core subsystems like ``Zend``, ``main`` and ``TSRM`` are also have maintainers.
I propose to create MAINTAINERS files in that directories with corresponding lists.
(V8 team uses OWNERS files, but this term may be not polite).

Reply:
Code owners is a common term and enables some GitHub features that might or might not be desirable (like automatic notification). https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

Reply:
The current `EXTENSIONS` file is also really outdated, for example:

```
EXTENSION:           date
PRIMARY MAINTAINER:  Derick Rethans <derick@php.net> (2005 - 2018)
```

Reply:
I think I can add a note to the RFC that this will also be used if present so if we add it in the future, it will be automatically taken from there.

Reply:
Extended that definition so hopefully it is better now.